### PR TITLE
docs: foundation layout written docs PTL-1505

### PR DESCRIPTION
**Pull request by matteematt**
Adds written docs for foundation layout

This has a lot of missing links, mostly to API docs. This will be added in a subsequent PTL ticket, and the links added

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/015_layout-management/layout_03_js_api.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+If you are only using the declarative API, and not using any dynamic integrations with JavaScript, then you shouldn't need to set the registration names of any items, as all the same items will be registered when you load a previously saved layout. If you are dynamically adding items as well, it is highly recommended to set the registration names of items manually. This makes it easier to figure out what is and is not registered.
+* When using the declarative API, use the `registration` attribute on the `<foundation-layout-item>` component.
+* When using the JavaScript API, set the `registration` optional parameter on the registered element config <!-- TODO link -->.
+:::
+
+## Saving and loading layout
+
+The JavaScript API can be used to save and load layout states manually. This only describes the state of the dynamic layout itself. It is the responsibility of each component within the layout to serialise its own state, if required. To enable autosaving the layout <!-- TODO link -->.
+
+## Methods
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/015_layout-management/layout_03_js_api.mdx:72]: This section will be fully linked to API docs when they're added in a subsequent ticket

File: docs/001_develop/03_client-capabilities/015_layout-management/index.mdx [hunk 2]
```diff
@@ -2,11 +2,149 @@
...
 
-Coming soon...
+import LayoutExample from './layout_example.js'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Layout Management
+
+Rapid layout is a module which facilitates the user to be able to perform actions such as dragging, resizing, and re-ordering windows.
+As a developer this module allows you to implement the above with low effort, or you can build upon it to create create powerful interactions such as dynamically adding/removing items, and saving/loading layouts.
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/015_layout-management/index.mdx:44]: the word create is here twice, create create

File: docs/001_develop/03_client-capabilities/015_layout-management/layout_05_binding_events_store.mdx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  name: 'my-element',
+  template: html<MyElement>`
+    <rapid-layout>
+      <rapid-layout-item>
+        <rapid-checkbox
+          ?checked=${sync((x) => x.isSelected, 'boolean')}
+        >Checkbox
+        </rapid-checkbox>
+      </rapid-layout-item>
+        <rapid-checkbox ?checked=${(x) => x.isSelected}></rapid-checkbox>
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/015_layout-management/layout_05_binding_events_store.mdx:107]: same here, missing the 'boolean'

File: docs/001_develop/03_client-capabilities/015_layout-management/layout_05_binding_events_store.mdx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      </rapid-layout-item>
+        <wrapper-two></wrapper-two>
+      <rapid-layout-item>
+    </rapid-layout>
+    `,
+})
+export class MyElement extends GenesisElement { }
+```
+
+There's still got a big issue here though - `isSelected` isn't linked between them anymore! Here is adding in some formal state management is required.
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/015_layout-management/layout_05_binding_events_store.mdx:158]: Reword this line, for grammatical purposes.

**Review by Gareth-Spencer-Genesis — APPROVED** [review]

**Review by matteematt — COMMENTED** [review]
@Gareth-Spencer-Genesis thanks for the review

**Comment by matteematt** [comment]
Getting this in due to @Gareth-Spencer-Genesis 's approval earlier, and the fact it's been hanging around a while. I don't think it's perfect but I think it's one that will be easier to review on the hosted website.

@wjhendry one issue I think we have here is that because we show off all of the complicated parts to it, it makes it seem quite difficult to use. There are aspects which are tricky, but most use cases are quite easy to do. It would be good if we can change the docs to make that clear

**Pull request merged**